client/app/components/queries/SchemaBrowser.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { react2angular } from 'react2angular';
+import PropTypes from 'prop-types';
+import { Icon } from 'antd';
+import { debounce } from 'lodash';
+import { $rootScope } from '@/services/ng';
+
+import { Schema } from '../proptypes';
+
+const isSchemaEmpty = schema => schema === undefined || !schema.length;
+
+// Return true if the search term is contained within the table name
+// or any of the table's columns.
+const doesTableContainSearchTerm = (searchTerm, table) => {
+  const regEx = new RegExp(searchTerm, 'gi');
+
+  // If the table name matches the search term we can return early.
+  if (regEx.test(table.name)) return true;
+
+  return !!table.columns.filter(column => regEx.test(column)).length;
+};
+
+const doCopy = (evt, hierarchy) => {
+  // eslint-disable-next-line
+  $rootScope.$broadcast('query-editor.command', 'paste', hierarchy.join('.'));
+  evt.stopPropagation();
+};
+
+class SchemaBrowser extends React.Component {
+  static propTypes = {
+    schema: Schema, // eslint-disable-line react/no-unused-prop-types
+    doRefresh: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    schema: null,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchTerm: '',
+      schema: [],
+      schemaInitialized: false,
+    };
+
+    // Candidate to be abstracted out to a set of config constants.
+    // Performs search filtering after the number of milliseconds set below.
+    this.doSearch = debounce(this.doSearch, 150);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.schema && this.props.schema.length && (prevState.schema.length !== this.props.schema.length)) {
+      // setState() is fine here within the right conditions.
+      // eslint-disable-next-line
+      if (!this.state.schemaInitialized) this.setState({ schema: this.props.schema });
+    }
+  }
+
+  handleSearch = (evt) => {
+    if (!evt.target.value) {
+      this.setState({ schema: this.props.schema, searchTerm: '' });
```

Comment:
`evt.target.value` is always string; here you should just update state and call `doSearch`, which should properly handle empty search term (return entire schema).